src/core/client/stream/tabs/Profile/DeletionRequest/DeletionRequestCalloutContainer.tsx

```diff
@@ -0,0 +1,94 @@
+import {
+  graphql,
+  useMutation,
+  withFragmentContainer,
+} from "coral-framework/lib/relay";
+import { Localized } from "fluent-react/compat";
+import React, { FunctionComponent, useCallback } from "react";
+
+import { useCoralContext } from "coral-framework/lib/bootstrap";
+import {
+  Button,
+  CallOut,
+  Flex,
+  HorizontalGutter,
+  Icon,
+  Typography,
+} from "coral-ui/components";
+
+import { DeletionRequestCalloutContainer_viewer } from "coral-stream/__generated__/DeletionRequestCalloutContainer_viewer.graphql";
+
+import CancelAccountDeletionMutation from "coral-stream/mutations/CancelAccountDeletionMutation";
+
+import styles from "./DeletionRequestCalloutContainer.css";
+
+interface Props {
+  viewer: DeletionRequestCalloutContainer_viewer;
+}
+
+const DeletionRequestCalloutContainer: FunctionComponent<Props> = ({
+  viewer,
+}) => {
+  const cancelDeletionMutation = useMutation(CancelAccountDeletionMutation);
+  const cancelDeletion = useCallback(() => {
+    cancelDeletionMutation();
+  }, [cancelDeletionMutation]);
+
+  const { locales } = useCoralContext();
+
+  const deletionDate = viewer.scheduledDeletionDate
+    ? Intl.DateTimeFormat(locales, {
+        year: "numeric",
+        month: "numeric",
+        day: "numeric",
+        hour: "numeric",
+        minute: "numeric",
+        second: "numeric",
+      }).format(new Date(viewer.scheduledDeletionDate))
+    : null;
+
+  return (
+    <>
+      {deletionDate && (
```

Comment:
To simplify, I'd suggest prior to the `const deletionDate = viewer.scheduledDeletionDate` line, you just do:

```js
if (!viewer.scheduledDeletionDate) {
  return null;
}
```

Then all the conditional logic after that line goes away.